### PR TITLE
docs(FEC-12629): Add "How to write plugin" documentation

### PR DESCRIPTION
**Pull request by JonathanTGold**
### Description of the Changes

Add How to write plugin documentation

And create a [plugin boilerplate ](https://github.com/kaltura/playkit-js-plugin-example)

solves: FEC-12629    FEC-12544

### CheckLists

- [ ] changes have been done against master branch, and PR does not conflict
- [ ] new unit / functional tests have been added (whenever applicable)
- [ ] test are passing in local environment
- [ ] Travis tests are passing (or test results are not worse than on master branch :))
- [x] Docs have been updated

### CODE REVIEW

File: docs/guides.md [hunk 1]
```diff
@@ -16,7 +16,7 @@
 - [User Preferences](./user-preferences.md)
 - [Controlling Errors](errors.md)
 - [Debugging & Troubleshooting](./debugging.md)
-- [How to Build a Plugin](./how-to-build-a-plugin.md)
+- [Writing-a-Plugin](./writing-a-plugin.md)
```
Comment (SivanA-Kaltura) [inline docs/guides.md:19]:
```suggestion
- [Writing a Plugin](./writing-a-plugin.md)
```

File: docs/writing-a-plugin.md [hunk 2]
```diff
@@ -241,72 +246,32 @@ We'll use both Vanilla JavaScript & ES6 examples.
...
-MyPlugin.prototype.destroy = function () {};
-
-MyPlugin.prototype.reset = function () {};
-```
-
-### _ES6_
-
 ```typescript
-class MyPlugin extends KalturaPlayer.core.BasePlugin {
+import {BasePlugin} from 'kaltura-player-js';
```
Comment (SivanA-Kaltura) [inline docs/writing-a-plugin.md:272]: I would use the original.
If you use import syntax you also need to explain how we use extern in webpack config to link to the specific player/preact instance, but the customer might not be using webpack at all.

**Review by SivanA-Kaltura — CHANGES_REQUESTED** [review]

File: docs/writing-a-plugin.md (hunk 2 above)
Comment (JonathanTGold) [reply docs/writing-a-plugin.md:272]: you don’t need Webpack in order to use `import` in ES6 you just need set the script to `type=module` as i did

Comment (SivanA-Kaltura) [reply docs/writing-a-plugin.md:272]: but you want to import from the local player, right ?

**Review by giladna — APPROVED** [review]

**Pull request merged**